contracts/TracerPerpetualSwaps.sol

```diff
@@ -487,31 +491,35 @@ contract TracerPerpetualSwaps is
         uint256 accountLastUpdatedIndex = balances[account].lastUpdatedIndex;
         uint256 currentGlobalFundingIndex = pricingContract
         .currentFundingIndex();
+        Balances.Account storage accountBalance = balances[account];
 
-        // Only settle account if its last updated index was before the current global index
-        if (accountLastUpdatedIndex < currentGlobalFundingIndex) {
-            /*
-             Get current and global funding statuses
-             Note: global rates reference the last fully established rate (hence the -1), and not
-             the current global rate. User rates reference the last saved user rate
-            */
+        // if this user has no positions, bring them in sync
+        if (accountBalance.position.base == 0) {
+            // set to the last fully established index
+            accountBalance.lastUpdatedIndex = currentGlobalFundingIndex;
+            accountBalance.lastUpdatedGasPrice = IOracle(gasPriceOracle)
+            .latestAnswer();
+        } else if (accountLastUpdatedIndex + 1 < currentGlobalFundingIndex) {
+            // Only settle account if its last updated index was before the last established
+            // global index this is since we reference the last global index
+            // Get current and global funding statuses
+            uint256 lastEstablishedIndex = currentGlobalFundingIndex - 1;
+            // Note: global rates reference the last fully established rate (hence the -1), and not
+            // the current global rate. User rates reference the last saved user rate
             Prices.FundingRateInstant memory currGlobalRate = pricingContract
-            .getFundingRate(pricingContract.currentFundingIndex() - 1);
+            .getFundingRate(lastEstablishedIndex);
```

Comment:
What if you are settling() on `currentGlobalFundingIndex == 0`? Then `accountLastUpdatedIndex == 0`, so neither of these branches will be entered.

Also, does this mean there will be a deadlock, since on `deposit(...)`, when a user will have base == 0, the pricing contract's global funding index doesn't get updated (I don't think), so I don't see how a user could enter a branch that allows pricing to update it's funding index if you get what I mean.

I think I'm probably misunderstanding some of the code here but just wanted to make sure :bhutan:

Reply:
@BenjaminPatch yep when both `currentGlobalFundingIndex == 0` and `accountLastUpdatedIndex == 0` that is expected behaviour, settle should not do anything unless your index is lagging behind the global funding index.

In terms of deadlocking, the global index only gets updated on trades, not on settle calls itself. So you are right in saying anyone that deposits within the first hour won't update the global index, but as soon as someone executed a trade in the second hour, the global index will get updated.

Most of this logic occurs in `pricing.sol` and is triggered by the `pricing.recordTrade()` function